qiskit_dynamics/signals/signals.py

```diff
@@ -299,39 +297,25 @@ def __init__(
         """
         self._dt = dt
 
-        samples = Array(samples)
+        samples = unp.asarray(samples)
 
         if len(samples) == 0:
-            zero_pad = np.array([0])
+            zero_pad = unp.array([0])
         else:
-            zero_pad = np.expand_dims(np.zeros_like(Array(samples[0])), 0)
-        self._padded_samples = np.append(samples, zero_pad, axis=0)
+            zero_pad = unp.expand_dims(unp.zeros_like(unp.asarray(samples[0])), 0)
+        self._padded_samples = unp.append(samples, zero_pad, axis=0)
 
         self._start_time = start_time
 
         # define internal envelope function
-        if samples.backend == "jax":
-
-            def envelope(t):
-                t = Array(t).data
-                idx = jnp.clip(
-                    jnp.array((t - self._start_time) // self._dt, dtype=int),
-                    -1,
-                    len(self.samples),
-                )
-                return self._padded_samples[idx]
-
-        else:
-
-            def envelope(t):
-                t = Array(t).data
-                idx = np.clip(
-                    np.array((t - self._start_time) // self._dt, dtype=int),
+        def envelope(t):
+            t = unp.asarray(t)
+            idx = unp.clip(
+                    unp.array((t - self._start_time) // self._dt, dtype=int),
                     -1,
                     len(self.samples),
                 )
-                return self._padded_samples[idx]
-
+            return self._padded_samples[idx]
```

Comment:
We need to improve this function. Because of it, cannot pass tests.

Reply:
What is the actual problem here? What is failing?